### PR TITLE
added Pagination, Sorting, and Swagger to the existing controllers

### PR DESCRIPTION
**Pull request by HAIOVYI**

### CODE REVIEW

**Review by MolchanovAlexander — CHANGES_REQUESTED** [review]
need some changes

File: src/main/java/mate/academy/mybookshop/controller/BookController.java [hunk 1]
```diff
@@ -15,18 +17,24 @@
...
     private final BookService bookService;
 
     @GetMapping
-    public List<BookDto> findAll() {
-        return bookService.findAll();
+    @Operation(summary = "Get all books", description = "Get a list of all available books")
+    public List<BookDto> findAll(@RequestParam(defaultValue = "0") int page,
+                                 @RequestParam(defaultValue = "10") int size,
+                                 @RequestParam(defaultValue = "id") String sortBy,
+                                 @RequestParam(defaultValue = "asc") String sortOrder) {
```
[the remark this answers is not in the document]
Comment (HAIOVYI) [reply src/main/java/mate/academy/mybookshop/controller/BookController.java]: agree better to use Pagable

File: src/main/java/mate/academy/mybookshop/controller/BookController.java [hunk 2]
```diff
@@ -15,18 +17,24 @@
...
+        return bookService.findAll(page, size, sortBy, sortOrder);
     }
 
     @GetMapping("/{id}")
```
[the remark this answers is not in the document]
Comment (HAIOVYI) [reply src/main/java/mate/academy/mybookshop/controller/BookController.java:37]: agree

File: src/main/java/mate/academy/mybookshop/service/BookService.java [hunk 3]
```diff
@@ -10,7 +10,7 @@ public interface BookService {
 
     BookDto findById(Long id);
 
-    List<BookDto> findAll();
+    List<BookDto> findAll(int page, int size, String sortBy, String sortOrder);
```
[the remark this answers is not in the document]
Comment (HAIOVYI) [reply src/main/java/mate/academy/mybookshop/service/BookService.java:15]: agree better to use Pagable

**Pull request merged**